Sources/StreamVideo/WebRTC/v2/StateMachine/Stages/WebRTCCoordinator+Joining.swift

```diff
@@ -0,0 +1,377 @@
+            sfuAdapter: SFUAdapter,
+            isFastReconnecting: Bool,
+            publisher: RTCPeerConnectionCoordinator?
+        ) async throws -> String {
+            let subscriberSessionDescription: String
+
+            if
+                isFastReconnecting,
+                let subscriber = await coordinator.stateAdapter.subscriber
+            {
+                let offer = try await subscriber.createOffer()
+                subscriberSessionDescription = offer.sdp
+
+            } else {
+                subscriberSessionDescription = try await RTCTemporaryPeerConnection(
+                    sessionID: coordinator.stateAdapter.sessionID,
+                    peerConnectionFactory: coordinator.stateAdapter.peerConnectionFactory,
+                    configuration: coordinator.stateAdapter.connectOptions.rtcConfiguration,
+                    sfuAdapter: sfuAdapter,
+                    videoOptions: coordinator.stateAdapter.videoOptions,
+                    localAudioTrack: publisher?.localTrack(of: .audio) as? RTCAudioTrack,
+                    localVideoTrack: publisher?.localTrack(of: .video) as? RTCVideoTrack
+                ).createOffer().sdp
+            }
+            return subscriberSessionDescription
+        }
+
+        private func join(
+            coordinator: WebRTCCoordinator,
+            sfuAdapter: SFUAdapter,
+            mergeWithLocalParticipants: Bool
+        ) async throws {
+            if let eventObserver = context.sfuEventObserver {
+                eventObserver.sfuAdapter = sfuAdapter
+            } else {
+                context.sfuEventObserver = .init(
+                    sfuAdapter: sfuAdapter,
+                    stateAdapter: coordinator.stateAdapter
+                )
+            }
+
+            let joinResponse = try await sfuAdapter
+                .publisher(eventType: Stream_Video_Sfu_Event_JoinResponse.self)
+                .nextValue(timeout: 10)
+            await context.previousSFUAdapter?.disconnect()
+            sfuAdapter.sendHealthCheck()
+
+            context.fastReconnectDeadlineSeconds = TimeInterval(
+                joinResponse.fastReconnectDeadlineSeconds
+            )
+
+            _ = try await sfuAdapter
+                .$connectionState
+                .filter {
+                    switch $0 {
+                    case .connected:
+                        return true
+                    default:
+                        return false
+                    }
+                }
+                .nextValue(timeout: 10)
+
+            var participants = joinResponse
+                .callState
+                .participants
+                .map { $0.toCallParticipant() }
+                .reduce(into: [String: CallParticipant]()) { partialResult, participant in
+                    partialResult[participant.sessionId] = participant
+                }
+
+            // - WORKAROUND - TO BE REMOVED - BEGIN
+            if mergeWithLocalParticipants {
```

Comment:
can you explain this in more details?

Reply:
That was a workaround needed because the SFU wasn't returning us the latest participant state whenever we migrated. That has now been fixed and thus the workaround has been removed.